src/v/pandaproxy/test/kafka_client_cache.cc

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "pandaproxy/kafka_client_cache.h"
+
+#include "config/node_config.h"
+#include "config/rest_authn_endpoint.h"
+#include "kafka/client/configuration.h"
+#include "pandaproxy/client_cache_error.h"
+#include "pandaproxy/types.h"
+
+#include <seastar/core/sleep.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/unit_test.hpp>
+
+#include <chrono>
+#include <vector>
+
+using namespace std::chrono_literals;
+
+namespace pandaproxy {
+class test_client_cache : public kafka_client_cache {
+public:
+    test_client_cache(
+      size_t max_size, std::vector<config::broker_authn_endpoint> kafka_api)
+      : kafka_client_cache(
+        to_yaml(kafka::client::configuration{}, config::redact_secrets::no),
+        max_size,
+        std::move(kafka_api),
+        (1000ms).count()) {}
+
+    test_client_cache(size_t max_size)
+      : test_client_cache(
+        max_size, std::move(config::node().kafka_api.value())) {}
+
+    client_ptr fetch(credential_t user) {
+        return kafka_client_cache::fetch(user);
+    }
+
+    void insert(credential_t user, client_ptr client) {
+        kafka_client_cache::insert(user, client);
+    }
+
+    client_ptr
+    make_client(credential_t user, config::rest_authn_type authn_type) {
+        return kafka_client_cache::make_client(user, authn_type);
+    }
+};
+} // namespace pandaproxy
+
+namespace pp = pandaproxy;
+
+SEASTAR_THREAD_TEST_CASE(test_cache_make_client) {
+    size_t max_size{10};
+    auto kafka_api = std::move(config::node().kafka_api.value());
+    kafka_api.clear();
+    kafka_api.push_back(config::broker_authn_endpoint{
+      .address = net::unresolved_address("127.0.0.1", 9092),
+      .authn_method = config::broker_authn_method::sasl});
+    pp::test_client_cache client_cache{max_size, std::move(kafka_api)};
```

Comment:
Moving from `config::node()` seems either dangerous or unnecessary, can you explain what your intention is?

Reply:
I actually intended to make a copy of the kafka_api above.
I'll change this line to pass kafka_api by value instead.